### PR TITLE
chore(django): add host/team_id to logs

### PR DESCRIPTION
**Pull request by hazzadous**
This is so we can see if anyone is sending us requests not to
app.posthog.com on Cloud US.

See https://posthog.slack.com/archives/C01MM7VT7MG/p1677596735032759 for context.

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Review by danielxnj — APPROVED** [review]

File: posthog/middleware.py [hunk 1]
```diff
@@ -370,3 +370,53 @@ def __call__(self, request: HttpRequest):
...
+    get_response: Callable[[HttpRequest], HttpResponse]
+) -> Callable[[HttpRequest], HttpResponse]:
+    """
+    This middleware adds the team_id to the logging context if it exists. Note
+    that this should be added after we have performed authentication, as we
+    need the user to be authenticated to get the team_id.
+    """
+
+    def middleware(request: HttpRequest) -> HttpResponse:
+        if request.user.is_authenticated:
```
Comment (pauldambra) [inline posthog/middleware.py:416]: I remember we had some problems with the rate limiting code from assuming team id was present...

So, we could be more cautious reading the user and its team...

But non-blocking - I might be bike-shedding

**Review by pauldambra — APPROVED** [review]
:shipit:

File: posthog/middleware.py (hunk 1 above)
Comment (hazzadous) [reply posthog/middleware.py:416]: Hah good catch 🙏 

I hope the tests caught something if it's an issue, will check.

Comment (hazzadous) [reply posthog/middleware.py:416]: Do you happen to have a link to the limiting PR that had the issue?

Comment (pauldambra) [reply posthog/middleware.py:416]: It lead to this method https://github.com/PostHog/posthog/blob/d196ea53cc7971c6ed1f7cff1fdae0682fe2c861/posthog/rate_limit.py#L53

Comment (pauldambra) [reply posthog/middleware.py:416]: Ah, it was _views_ that don't have a team id https://github.com/PostHog/posthog/pull/12006

**Pull request merged**